pkg/service/driver.go

```diff
@@ -23,17 +23,32 @@ type KubevirtCSIDriver struct {
 	Client             kubevirt.Client
 }
 
-func NewKubevirtCSIDriver(infraClusterClient kubevirt.Client, infraClusterNamespace string, infraClusterLabels map[string]string, nodeID string) *KubevirtCSIDriver {
-	d := KubevirtCSIDriver{
-		IdentityService: &IdentityService{
-			infraClusterClient: infraClusterClient,
-		},
-		ControllerService: &ControllerService{
+func NewKubevirtCSIDriver(infraClusterClient kubevirt.Client, infraClusterNamespace string, infraClusterLabels map[string]string, nodeID string, runNodeService bool, runControllerService bool) *KubevirtCSIDriver {
+	var nodeService *NodeService
+	var controllerService *ControllerService
+
+	if (runNodeService) {
+		nodeService = NewNodeService(infraClusterClient, nodeID) 
+	} else {
+		nodeService = nil
+	}
+	
+	if(runControllerService) {
+		controllerService = &ControllerService{
 			infraClient:           infraClusterClient,
 			infraClusterNamespace: infraClusterNamespace,
 			infraClusterLabels:    infraClusterLabels,
+		}
+	} else {
+		controllerService = nil
+	}
+
+	d := KubevirtCSIDriver{
+		IdentityService: &IdentityService{
+			infraClusterClient: infraClusterClient,
 		},
```

Comment:
This will cause the probe operation to fail on the tenant cluster because the tenant does not have access to the infra cluster once we split the controller and node logic.

I've taken your PR and revised it here https://github.com/kubevirt/csi-driver/pull/45/ with probe logic that should work for both the infra and tenant cluster. I set you as co-author since i used a lot of the structure you already had in this pr.